### PR TITLE
fix: ensure that tini is using the correct CPU architecture

### PR DESCRIPTION
**Pull request by dduportal**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- ~[ ] Link to relevant issues in GitHub or Jira~
- ~[ ] Link to relevant pull requests, esp. upstream and downstream changes~
- ~[ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue~

This  PR follows up a discussion on the IRC channel #jenkins-infra were @MarkEWaite realized that some of the new Jenkins images were not working as expected: for all the non-debian images on ARM64, the container was not starting with the dreaded `exec user format` which indicates a binary uncomapitble with the current CPU architecture.

The fix consists on using the variable `TARGETARCH` (populated by Docker Buildx) to determine the CPU architecture and use it to get the correct binary.

I tested amd64 and arm64 quickly: we have to be sure that it also works for PPC and s390x.

=> Why did this issue did not trigger an error in the build? it's because we are using QEMU everywhere (so the amd64 binary was executed without error).

This PR also remove the former multiarch `Dockerfile` as they are not used anymore (to anyone whom contributed to these deleted files: thanks a lot because you paved the way for a fully working multi arch!)

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

File: multiarch/Dockerfile.alpine [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Placeholder for the specified arch that gets parsed in the publish-experimental.sh script.
 FROM BASEIMAGE
 
+ARG TARGETARCH
```
Comment (timja) [inline multiarch/Dockerfile.alpine:4]: this file should be deleted, it's not used

File: multiarch/Dockerfile.debian [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Placeholder for the specified arch that gets parsed in the publish-experimental.sh script.
 FROM BASEIMAGE
 
+ARG TARGETARCH
```
Comment (timja) [inline multiarch/Dockerfile.debian:4]: same here

File: multiarch/Dockerfile.slim [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Placeholder for the specified arch that gets parsed in the publish-experimental.sh script.
 FROM BASEIMAGE
 
+ARG TARGETARCH
```
Comment (timja) [inline multiarch/Dockerfile.slim:4]: and here

**Review by timja — APPROVED** [review]
quote changes aren't needed but shouldn't cause any harm

**Review by timja — APPROVED** [review]

**Review by timja — APPROVED** [review]
I've verified this on both s390x and ppc64le.

ppc64le fix in https://github.com/krallin/tini/releases/tag/v0.19.0 is required for tini to work on it, I pushed a commit to sort that out

**Pull request merged**